metadata.go

```diff
@@ -0,0 +1,86 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ttrpc
+
+import "context"
+
+// Metadata represents the key-value pairs (similar to http.Header) to be passed to ttrpc server from a client.
+type Metadata map[string]StringList
```

Comment:
This type is not well supported by protobuf.

Reply:
Also, this will disallow repeat values for headers, which will diverge support with grpc on http/2.